crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,364 @@
+    let Some(parent) = JsParenthesizedExpression::cast(parent) else {
+        return exp;
+    };
+    get_parenthesized_parent(AnyJsExpression::from(parent))
+}
+
+fn get_parenthesized_child(exp: &AnyJsExpression) -> AnyJsExpression {
+    let AnyJsExpression::JsParenthesizedExpression(parenthesized) = exp else {
+        return exp.clone();
+    };
+    let Some(child) = parenthesized.expression().ok() else {
+        return exp.clone();
+    };
+    get_parenthesized_child(&child)
+}
+
+impl Rule for NoUselessLengthCheck {
+    type Query = Ast<JsLogicalExpression>;
+    type State = (ErrorType, Replacer);
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let Some(operator) = node.operator_token().ok() else {
+            return Vec::new();
+        };
+        // node must not be a child of a logical expression
+        if is_logical_exp_descendant(&AnyJsExpression::from(node.clone()), operator.kind()) {
+            return Vec::new();
+        }
+
+        let mut fixable_list: Vec<(ErrorType, Replacer)> = Vec::new();
+
+        for err_type in [
+            ErrorType::UselessLengthCheckWithEvery,
+            ErrorType::UselessLengthCheckWithSome,
+        ] {
+            let mut comparing_zeros: HashMap<String, Vec<Replacer>> = HashMap::new();
+            let mut array_tokens_used_api: HashSet<String> = HashSet::new();
+            let search_result = search_logical_exp(
+                &AnyJsExpression::from(node.clone()),
+                None,
+                &err_type,
+                &mut comparing_zeros,
+                &mut array_tokens_used_api,
+            );
+            if search_result.is_some() {
+                for array_token in array_tokens_used_api {
+                    if let Some(replacers) = comparing_zeros.get(&array_token) {
+                        for replacer in replacers {
+                            fixable_list.push((err_type.clone(), replacer.clone()));
+                        }
+                    }
+                }
+            }
+        }
+        fixable_list
+    }
+
+    fn diagnostic(
+        _ctx: &RuleContext<Self>,
+        (error_type, (_, _, error_range)): &Self::State,
+    ) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                error_range,
+                markup! {
+                    "This length check is unnecessary."
+                },
+            )
+            .note(
+                match error_type {
+                    ErrorType::UselessLengthCheckWithEvery => markup! {
+                        "The empty check is useless as `Array#every()` returns `true` for an empty array."
+                    },
+                    ErrorType::UselessLengthCheckWithSome => markup! {
+                        "The non-empty check is useless as `Array#some()` returns `false` for an empty array."
```

Comment:
We can use `<Emphasis>` here to `Array#every()`, `true`, `Array#some()`, `false`